lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -244,7 +244,7 @@ def formatted_message_and_backtrace(colorizer)
         end
 
         def exception_backtrace
-          exception.backtrace || []
+          RSpec.configuration.backtrace_formatter.format_backtrace(exception.backtrace) || []
```

Comment:
I can see how in your case, it would be preferable to have it show the snippet from the spec file instead of the support file.  However, I don't think it makes sense to use `backtrace_exclusion_patterns` to exclude snippets -- as @JonRowe said, the config option is about backtraces, not snippets and having it affect both would be confusing and surprising.  Plus, if we can't find a stack frame in any of the non-excluded files, we would like to show the top line from the excluded files so that we show some kind of snippet.

As it turns out, we already have a configurable mechanism to control what code is chosen for the purpose of code snippets:

http://rspec.info/documentation/3.5/rspec-core/RSpec/Core/Configuration.html#project_source_dirs-instance_method

If you have a file of code that you don't want RSpec to consider to be part of the available source code from which to draw snippets, then configure `project_source_dirs` appropriately (or move the file to another directory).  By default, `project_source_dirs` is set to `["lib", "app", "spec"]`.  For your situation I would recommend moving the file containing the `waiter` method out of `spec/support` and into a directory like `spec_support` and then load it with `require_relative`.  Then RSpec won't consider it part of the body of source code for code snippets. And you can get the behavior you want.